cli/src/cli_util.rs

```diff
@@ -1017,15 +1025,29 @@ impl WorkspaceCommandHelper {
                 let mut iter = [commit0, commit1].into_iter().chain(iter);
                 let commits: Vec<_> = iter.by_ref().take(5).try_collect()?;
                 let elided = iter.next().is_some();
-                let hint = format!(
-                    r#"The revset "{revision_str}" resolved to these revisions:{eol}{commits}{ellipsis}"#,
-                    eol = "\n",
-                    commits = commits
-                        .iter()
-                        .map(|c| self.format_commit_summary(c))
-                        .join("\n"),
-                    ellipsis = elided.then_some("\n...").unwrap_or_default()
-                );
+                // Separate hint if the conflict is due to a branch conflict
+                let hint = if let Some(branch_name) = branch {
+                    format!(
+                        r#"The revset "{revision_str}" resolved to these revisions:{eol}{commits}{ellipsis}
+This error is due to a conflicted branch. To resolve this, try setting to which revision the branch points to with "jj branch set {branch_name} --revision <REVISION>""#,
+                        eol = "\n",
```

Comment:
@ilyagr, do you remember why you added this `eol` argument in #1189?

Reply:
I think it had to do with the fact that "\n" did not expand inside the raw string. So, I put `{eol}` inside the raw string.

One option for this I recently found is https://github.com/dtolnay/indoc, but maybe that's not for this PR.

Reply:
You can simply include a literal newline in the string, but maybe you wanted to avoid the unindented line. Yes, we could use `indoc` (I think we even used to use that somewhere).